core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCommonQueryBuilder.java

```diff
@@ -1891,9 +1895,7 @@ protected void appendGroupByClause(StringBuilder sbSelectFrom) {
         Set<String> clauses = new LinkedHashSet<String>();
         groupByManager.buildGroupByClauses(clauses);
 
-        int size = transformerGroups.size();
```

Comment:
It is a micro optimization, but why not help the compiler/interpreter and GC when possible? I generally agree with you that if the implementation isn't known that an enhanced-for loop is a better approach, but I don't consider an indexed loop to be unreadable. The implementation is controlled by us and such optimizations are useful in general. I like to avoid creating garbage i.e. the iterator object whenever possible.